src/main/java/io/confluent/connect/jdbc/dialect/GenericDatabaseDialect.java

```diff
@@ -0,0 +1,1753 @@
+                }
+                return clob.getSubString(1, (int) clob.length());
+              } finally {
+                if (isJdbc4) {
+                  free(clob);
+                }
+              }
+            }
+          }
+        };
+      case Types.NCLOB: {
+        return new ColumnConverter() {
+          @Override
+          public Object convert(ResultSet rs) throws SQLException, IOException {
+            Clob clob = rs.getNClob(col);
+            if (clob == null) {
+              return null;
+            } else {
+              try {
+                if (clob.length() > Integer.MAX_VALUE) {
+                  throw new IOException("Can't process NCLOBs longer than " + Integer.MAX_VALUE);
+                }
+                return clob.getSubString(1, (int) clob.length());
+              } finally {
+                if (isJdbc4) {
+                  free(clob);
+                }
+              }
+            }
+          }
+        };
+      }
+
+      // XML -> string
+      case Types.SQLXML: {
+        return new ColumnConverter() {
+          @Override
+          public Object convert(ResultSet rs) throws SQLException {
+            SQLXML xml = rs.getSQLXML(col);
+            return (xml != null ? xml.getString() : null);
+          }
+        };
+      }
+
+      case Types.NULL:
+      case Types.ARRAY:
+      case Types.JAVA_OBJECT:
+      case Types.OTHER:
+      case Types.DISTINCT:
+      case Types.STRUCT:
+      case Types.REF:
+      case Types.ROWID:
+      default: {
+        // These are not currently supported, but we don't want to log something for every single
+        // record we translate. There will already be errors logged for the schema translation
+        break;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Called when the object has been fully read and {@link Blob#free()} should be called.
+   *
+   * @param blob the Blob; never null
+   * @throws SQLException if there is a problem calling free()
+   */
+  protected void free(Blob blob) throws SQLException {
+    blob.free();
+  }
+
+  /**
+   * Called when the object has been fully read and {@link Clob#free()} should be called.
+   *
+   * @param clob the Clob; never null
+   * @throws SQLException if there is a problem calling free()
+   */
+  protected void free(Clob clob) throws SQLException {
+    clob.free();
+  }
+
+  @Override
+  public String buildInsertStatement(
+      TableId table,
+      Collection<ColumnId> keyColumns,
+      Collection<ColumnId> nonKeyColumns
+  ) {
+    ExpressionBuilder builder = expressionBuilder();
+    builder.append("INSERT INTO ");
+    builder.append(table);
+    builder.append("(");
+    builder.append(",", ExpressionBuilder.columnNames(), keyColumns, nonKeyColumns);
+    builder.append(") VALUES(");
+    builder.appendMultiple(",", "?", keyColumns.size() + nonKeyColumns.size());
+    builder.append(")");
+    return builder.toString();
```

Comment:
Here's an example of how the `ExpressionBuilder` is used. Here, `table` is a `TableId` and can be appended to the expression builder and automatically expanded into the fully-qualified quoted identifier. Line 1405 appends each key column and non-key column but delimited by a comma and properly quoted names (rather than fully-qualified identifiers).